### PR TITLE
feat: Audit and validation framework for CWR (RDCC-2056)

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2056

### Change description ###
feat: Audit and validation framework for CWR (RDCC-2056)

**Does this PR introduce a breaking change?** (check one with "x")
No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java [hunk 1]
```diff
@@ -64,12 +64,27 @@ private CaseWorkerConstants() {
 
     public static final String ORIGIN_EXUI = "EXUI";
 
+    public static final String UP_FAILURE_ROLES = "can't modify roles for user in UP";
+
+    public static final String STATUS_ACTIVE = "ACTIVE";
+
+    public static final String ALREADY_SUSPENDED_ERROR_MESSAGE = "user is suspended already";
+
+    public static final String UP_CREATION_FAILED = "Failed to crate in UP with response status %s";
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java:73]: Typo in the word create

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java [hunk 2]
```diff
@@ -64,12 +64,27 @@ private CaseWorkerConstants() {
...
+
+    public static final String REQUEST_FAILED_FILE_UPLOAD_JSR =
+        "Request completed with partial success. Some records failed during validation and were ignored.";
+
+    public static final String RECORDS_FAILED = "%s record(s) failed validation";
+
+    public static final String RECORDS_SUSPENDED = "%s  record(s) suspended";
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java:80]: Looks like there is an extra space before `record(s)`

**Review by SujeethKumarAri — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 3]
```diff
@@ -204,7 +204,8 @@ sonarqube {
                 "src/main/java/uk/gov/hmcts/reform/cwrdapi/client/domain/**," +
                 "src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/**," +
                 "src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/request/**," +
-                "src/main/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefApiApplication.java"
+                "src/main/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefApiApplication.java," +
+                "src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java"
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply build.gradle:209]: yes..we discussed exception list is not covrerd as update and supsion logic flow exception and will be removed in friday's final PR

[the remark this answers is not in the document]
Comment (SujeethKumarAri) [reply build.gradle:209]: I forgot that.. cheers

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Pull request merged**